lib/helpers/rule-test-harness.js

```diff
@@ -91,15 +103,24 @@ module.exports = function generateRuleTests({
 
     let linter;
 
+    function metaToObject(item) {
+      const fallbackMeta = JSON.parse(JSON.stringify(defaultMeta || {}));
+      if (typeof item !== 'object' || item === null) {
+        return fallbackMeta;
+      }
+      if (typeof item.meta === 'object') {
+        return item.meta;
+      } else {
+        return fallbackMeta;
+      }
+    }
+
     function parseMeta(item) {
-      let meta =
-        item !== undefined && typeof item === 'object' && item.meta
-          ? item.meta
-          : defaultMeta !== undefined
-          ? JSON.parse(JSON.stringify(defaultMeta))
-          : {};
-      meta.filePath = meta.filePath || 'layout.hbs';
-      meta.moduleId = meta.moduleId || 'layout';
+      let meta = metaToObject(item);
+      meta.filePath = meta.filePath || ConfigDefaults.filePath;
+      meta.moduleId = meta.moduleId || ConfigDefaults.moduleId;
+      meta.workingDir = meta.workingDir || ConfigDefaults.workingDir;
```

Comment:
I'd rather these use an `in` check instead of falseyness

```suggestion
      meta.filePath = 'filePath' in meta ? meta.filePath : ConfigDefaults.filePath;
      meta.moduleId = 'moduleId' in meta ? meta.moduleId : ConfigDefaults.moduleId;
      meta.workingDir = 'workingDir' in meta ? meta.workingDir : ConfigDefaults.workingDir;
```

Reply:
lol, this change cost us dozens of failed tests

Reply:
![image](https://user-images.githubusercontent.com/1360552/106914231-dd515080-6715-11eb-8af5-a8f3421ca775.png)

Reply:
FWIW, this `in` check makes it possible for us to deprecate specifying / accessing `moduleId`